Explorer/Assets/DCL/Navmap/NavmapSearchBarController.cs

```diff
@@ -69,13 +118,39 @@ private void ClearSearch()
             view.inputField.SetTextWithoutNotify("");
             searchResultPanelController.Hide();
             view.clearSearchButton.gameObject.SetActive(false);
-            ResetFloatingPanelStatus();
         }
 
-        private void ResetFloatingPanelStatus()
+        private void GetAndShowPreviousSearches()
+        {
+            cts = cts.SafeRestart();
+            previousSearches = GetPreviousSearches();
+            if (previousSearches.Length <= 0) return;
+
+            historyRecordPanelView.gameObject.SetActive(true);
+            historyRecordPanelView.SetHistoryRecords(previousSearches);
+        }
+
+        private void AddToPreviousSearch(string searchToAdd)
+        {
+            playerPrefsPreviousSearches = PlayerPrefs.GetString(PREVIOUS_SEARCHES_KEY);
+            previousSearches = string.IsNullOrEmpty(playerPrefsPreviousSearches) ? Array.Empty<string>() : playerPrefsPreviousSearches.Split('|');
+            switch (previousSearches.Length)
+            {
+                case > 0 when previousSearches[0] == searchToAdd:
+                    return;
+                case < 5:

```

Comment:
Can you give names to the magic numbers? It's not clear why it's happening this way at first glance